src/status_im/wallet/choose_recipient/core.cljs

```diff
@@ -172,7 +172,7 @@
                         (assoc-in message path address))
                       message
                       (map vector paths addresses))]))}}))
-    {:ui/show-error (i18n/label :t/wallet-invalid-address {:data uri}})))
+    (assoc db :ui/show-error (i18n/label :t/wallet-invalid-address {:data uri}))))
```

Comment:
its not about assoc it's just about right balance of parentheses

Reply:
`{:ui/show-error (i18n/label :t/wallet-invalid-address {:data uri})}`

Reply:
I guess I'm not seeing it.  It's compiling correctly, the error message displays as intended when an invalid address is shown, and the IDE isn't showing any issues with that branch of the `if-let`.

Reply:
yeah probably it will work, but it's wrong, you have to return a map of effects in this function, its enough to just return `{:ui/show-error (i18n/label :t/wallet-invalid-address {:data uri})}` map with :ui/show-error effect, but you return app db map, it's wrong , so my code suggestion was correct, but i confused  parentheses it's hard do make this properly here on github without IDE

Reply:
Understood, will update later today once I'm back on my PC and thanks for the explanation!

Reply:
Fixed per your edit.